src/main/java/org/wildfly/security/credential/store/impl/KeyStoreCredentialStore.java

```diff
@@ -655,6 +655,7 @@ public void remove(final String credentialAlias, final Class<? extends Credentia
                     }
                 }
             }
+            cache.remove(credentialAlias);
```

Comment:
But cannot be keystore in same minor cases changed, even through exception is thrown later? Would not be removing from cache safer here?